internal/idutils/external_id.go

```diff
@@ -18,26 +18,34 @@ package idutils
 
 import (
 	"encoding/base64"
-	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/coordinate"
+	"strings"
 )
 
-// GenerateExternalID generates the externalID for settings 2.0 objects based on the schema, and ID.
-// The result of the function is pure.
-// Max length for the external ID is 500
-func GenerateExternalID(schema, id string) string {
+// GenerateExternalID returns a string that can be used as an external ID for a Settings 2.0 object
+func GenerateExternalID(coordinate coordinate.Coordinate) string {
+	if coordinate.Type == "" || coordinate.ConfigId == "" {
+		panic("Settings schema id and config id needs to be set to generate a valid external id for settings 2.0 objects")
```

Comment:
can this actually happen in some random runtime situations? If not, then why validate against it.
If there's a way that happens then I'd prefer returning an error.